src/coredump/coredump.c

```diff
@@ -1087,40 +1096,67 @@ static int process_socket(int fd) {
                         goto finish;
                 }
 
-                /* The final zero-length datagram carries the file descriptor and tells us
+                /* The final zero-length datagram carries the file descriptors and tells us
                  * that we're done. */
                 if (n == 0) {
```

Comment:
so the logic is broken now, as you accept now that the first empty msg is received, and then some data blobs again, and then the second empty msg.

I think it would be much nicer if we would simply accept either 1 or 3 fds on a final zero length msg. That's perfectly compatible with the current logic, a natural extension, and allows no vagueness about the protocol.

Reply:
But what we accept is one aspect, second is what we send. 

During package updates/downgrades we may run into situation when message is generated by one version (older/newer ?) and processed by another version because in the meantime systemd-coredump got replaced. To cope with this I send two messages (first with 1 fd and second with 2 fds) and when processing I allow for at most two zero-length messages but so far I didn't check whether there are no data message in between them (they shouldn't be).